File: itv/IvanRC-ITV-Java/src/main/java/models/Vehiculo.java
package models;

public class Vehiculo {
    private final String modelo;
    private int kilometros;
    private final int añoMatriculacion;
    private final boolean apto;

    public Vehiculo(String modelo, int kilometros, int añoMatriculacion, boolean apto) {
        this.modelo = modelo;
        this.kilometros = kilometros;
        this.añoMatriculacion = añoMatriculacion;
        this.apto = apto;
    }

    public String getModelo() {
        return modelo;
    }

    public int getKilometros() {
        return kilometros;
    }

    public void setKilometros(int kilometros) {
        this.kilometros = kilometros;
    }

    public int getAñoMatriculacion() {
        return añoMatriculacion;
    }

    public boolean isApto() {
        return apto;
    }
}

File: itv/IvanRC-ITV-Funcional/src/main/kotlin/models/Vehiculo.kt
package models

import java.time.LocalDate

sealed class Vehiculo(
    val modelo: String,
    val kilometros: Int,
    val añoMatriculacion: Int,
    val Apto: Boolean
)
class Coche(
    modelo: String,
    kilometros: Int,
    añoMatriculacion: Int,
    Apto: Boolean,
    val numeroPuertas: Int
): Vehiculo(modelo, kilometros, añoMatriculacion, Apto){
    override fun toString(): String {
        return "Coche(modelo='$modelo', kilometros=$kilometros, añoMatriculacion=$añoMatriculacion, Apto=$Apto, numeroPuertas=$numeroPuertas)"
    }
}
class Moto(
    modelo: String,
    kilometros: Int,
    añoMatriculacion: Int,
    Apto: Boolean,
    val cilidrada: Int
): Vehiculo(modelo, kilometros, añoMatriculacion, Apto){
    override fun toString(): String {
        return "Moto(modelo='$modelo', kilometros=$kilometros, añoMatriculacion=$añoMatriculacion, Apto=$Apto, cilindrada=$cilidrada)"
    }
}